### ISSUE TITLE
grouped_execution_for_eligible_table_scans causes non-partition tables to fail

### ISSUE THREAD

**Issue by highker** [prestodb/presto#13322]
To reproduce:
```
presto:default> set session grouped_execution_for_eligible_table_scans=true;
SET SESSION
presto:default> SELECT * FROM tpch.tiny.orders;
Query 20190831_222335_00013_gvva8 failed: Internal error
java.lang.IllegalArgumentException
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:121)
	at com.facebook.presto.sql.planner.PlanFragmenter$GroupedExecutionProperties.<init>(PlanFragmenter.java:1141)
	at com.facebook.presto.sql.planner.PlanFragmenter$GroupedExecutionTagger.visitTableScan(PlanFragmenter.java:1066)
	at com.facebook.presto.sql.planner.PlanFragmenter$GroupedExecutionTagger.visitTableScan(PlanFragmenter.java:885)
	at com.facebook.presto.spi.plan.TableScanNode.accept(TableScanNode.java:150)
	at com.facebook.presto.sql.planner.PlanFragmenter.analyzeGroupedExecution(PlanFragmenter.java:210)
	at com.facebook.presto.sql.planner.PlanFragmenter.analyzeGroupedExecution(PlanFragmenter.java:244)
	at com.facebook.presto.sql.planner.PlanFragmenter.createSubPlans(PlanFragmenter.java:173)
	at com.facebook.presto.execution.SqlQueryExecution.doAnalyzeQuery(SqlQueryExecution.java:437)
	at com.facebook.presto.execution.SqlQueryExecution.analyzeQuery(SqlQueryExecution.java:410)
	at com.facebook.presto.execution.SqlQueryExecution.startExecution(SqlQueryExecution.java:354)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:514)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1135)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:844)
```

cc: @shixuan-fan @wenleix

**Comment by shixuan-fan** [comment]
@arhimondr also reported this to me last week. I'll take a look.

**Comment by shixuan-fan** [comment]
Closed via #13329
